conmon-rs/server/src/lib.rs

```diff
@@ -106,27 +120,38 @@ impl Server {
         .await?
     }
 
-    async fn start_sigterm_handler<T: AsRef<Path>>(
+    async fn start_signal_handler<T: AsRef<Path>>(
+        reaper: Arc<child_reaper::ChildReaper>,
         socket: T,
         shutdown_tx: oneshot::Sender<()>,
     ) -> Result<()> {
         let mut sigterm = signal(SignalKind::terminate())?;
         let mut sigint = signal(SignalKind::interrupt())?;
+        let handled_sig: Signal;
 
         tokio::select! {
             _ = sigterm.recv() => {
                 info!("Received SIGTERM");
+                handled_sig = Signal::SIGTERM;
             }
             _ = sigint.recv() => {
                 info!("Received SIGINT");
+                handled_sig = Signal::SIGINT;
             }
         };
 
         let _ = shutdown_tx.send(());
+
+        // TODO FIXME Ideally we would drop after socket file is removed,
+        // but the removal is taking longer than 10 seconds, indicating someone
+        // is keeping it open...
+        reaper.kill_children(handled_sig)?;
```

Comment:
BTW do we need this versus `PRCTL_PR_SET_PDEATHSIG`?  See e.g. https://github.com/containers/containers-image-proxy-rs/pull/16

Reply:
pdeathsignal just kills conmon-rs and I'm not sure why. maybe something interacting with tokio?